internal/helpers/secrets.go

```diff
@@ -89,6 +200,10 @@ func SyncSecret(ctx context.Context, client ctrlclient.Client, obj ctrlclient.Ob
 		Name:      meta.Destination.Name,
 	}
 
+	if err := common.ValidateObjectKey(key); err != nil {
+		return fmt.Errorf("empty name for destination Secret, err=%w", err)
```

Comment:
Ni: I believe `ValidateObjectKey` will return `err` if either name/namespaces are empty string, suggest changing the wording a bit.
```suggestion
		return fmt.Errorf("empty name or namespace for destination Secret, err=%w", err)
```

Right now this will output something like:
`empty name (or namespace) for destination Secret, err=invalid objectKey` which maybe seems a bit redundant, but change it how you see fit, if you see fit :)

Reply:
Hmm, we probably want to make the message more generic, and ensure that the wrapped error provides the relevant bits?

Reply:
Fixed with d9e218884470d5ff0394f56b84657325ed443131